pkg/didcomm/protocol/didexchange/persistence.go

```diff
@@ -103,14 +113,65 @@ func (c *ConnectionRecorder) GetConnection(connectionID string) (*ConnectionReco
 	return &ConnectionRecord{State: string(name)}, nil
 }
 
-// invitationKey computes key for invitation object
-func invitationKey(verKey string) (string, error) {
-	storeKey, err := computeHash([]byte(verKey))
+// saveConnectionRecord saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) saveConnectionRecord(invitation *Invitation,
+	msg *message) error {
+	connRecord := &ConnectionRecord{ThreadID: msg.ThreadID,
+		ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey:          invitation.RecipientKeys, ConnectionID: msg.ConnectionID, State: msg.State}
+
+	k, err := connectionKey(msg.ConnectionID)
+	if err != nil {
+		return err
+	}
+
+	bytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	return c.store.Put(k, bytes)
+}
+
+// saveProtocolID saves the hash of the protocol id which is a did and thread id as an ID and connection id as value.
+func (c *ConnectionRecorder) saveProtocolID(myDID, thid, connectionID string) error {
+	protocolID := myDID + thid
```

Comment:
Better to validate(empty check) myDID and thid as these are the composite keys.

Reply:
done